File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    const answer = [];
    const window = [];
    
    for(let i=0; i<score.length; i++){
        if(window.length < k) window.push(score[i])
        else{
            let min = Math.min(...window)
            if(min < score[i]){
                let minIdx = window.indexOf(min)
                window.splice(minIdx,1,score[i])
            }
        }
        answer.push(Math.min(...window))
    }
    return answer;
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    answer = [] #최하위 값 모음
    window = [] #명예의 전당
    
    for x in score:
        if len(window) < k:
            window.append(x)
        else: #자리 다참
            if min(window) < x:
                window.remove(min(window))
                window.append(x)
        answer.append(min(window))
        
    return answer